inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -159,6 +144,230 @@ void MKLDNNGraphOptimizer::ApplyImplSpecificGraphOptimizations(MKLDNNGraph &grap
     graph.RemoveDroppedEdges();
 }
 
+void MKLDNNGraphOptimizer::FuseConvolutionAndBias(MKLDNNGraph &graph) {
+    auto& graphNodes = graph.GetNodes();
+
+    auto isSutableParentNode = [](MKLDNNNodePtr node) {
+        return node->getType() == Convolution &&
+               node->getChildEdges().size() == 1 &&
+               node->getFusedWith().empty();
+    };
+
+    auto isSutableChildNode = [&](MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
+        if (childNode->getAlgorithm() != EltwiseAdd || !childNode->getFusedWith().empty() || childNode->getParentEdges().size() != 2)
+            return false;
+
+        auto biasNode = childNode->getParentEdgesAtPort(1)[0]->getParent();
+        if (!biasNode->isConstant() || biasNode->getChildEdges().size() != 1)
```

Comment:
\+ check biasNode->getType() != Input

Reply:
Just realized that Bias is not required to be constant at all. We just reconnect arbitrary input subgraph from child Add node as third input of parent Convolution. Since Convolution can correctly handle dynamic subgraphs on each input it shouldn't be the limitation.